tools/vpc-flowlogs-enforcer/terraform/templates/cloud_function/main.py

```diff
@@ -0,0 +1,174 @@
+"""
+ Copyright 2020 Google LLC. This software is provided as-is, without warranty
+ or representation for any use or purpose. Your use of it is subject to your
+ agreement with Google.
+"""
+import logging
+import traceback
+import base64
+import json
+import re
+import os
+import sys
+import time
+from datetime import datetime, timezone
+from dateutil import parser
+from googleapiclient import discovery
+
+
+def enable_net_logs(event, context):
+    """
+  Enable flow logs in a subnetwork using the log configuration provided in the
+  LOG_CONFIG environment variable.
+  """
+    default_log_config = '{"aggregationInterval" : "INTERVAL_5_SEC", ' \
+      '"flowSampling" : "0.75", "metadata" : "INCLUDE_ALL_METADATA"}'
+    vpc_log_config = os.getenv('LOG_CONFIG', default_log_config)
+
+    if not 'data' in event:
+        logging.error('no data found. Ignoring event.')
+        return
+
+    pubsub_message = base64.b64decode(event['data']).decode('utf-8')
+    json_msg = None
+    try:
+        json_msg = json.loads(pubsub_message)
+    except ValueError:  # includes simplejson.decoder.JSONDecodeError
+        logging.error('invalid pubsub message: %s', pubsub_message)
+        return
+
+    logs_enabled = False
+    project = region = subnetwork = None
+
+    # Initialize the compute API
+    service = discovery.build('compute', 'v1', cache_discovery=False)
+
+    # Stackdriver change notification
+    if 'resource' in json_msg and 'type' in json_msg['resource'] and json_msg[
```

Comment:
I would prefer having this code split out into separate smaller functions, which take a json as input and can then be easily tested